Export screening results to csv file

azywot opened the pull request:
Add `Save CSV` option - create & save a csv file, cache a file for 60s and delete it 
![obraz](https://github.com/zuzg/drug-screening/assets/82370491/74279c60-5396-489b-8e7d-561e77753a4e)

csv file structure serving as an input to `Correlation Analysis`:
![obraz](https://github.com/zuzg/drug-screening/assets/82370491/186e3f86-287d-46ff-85cc-266ccdaa8b0a)

Any suggestions are welcome 😊

Tremirre commented on dashboard/pages/screening/callbacks.py:
```diff
@@ -345,6 +371,52 @@ def on_z_score_range_update(n_clicks, figure, range):
+    )
+
+    echo_bmg_combined_df = reorder_bmg_echo_columns(echo_bmg_combined_df)
+    csv_buffer = echo_bmg_combined_df.to_csv(
```
Why not send the csv directly to the user instead of saving and deleting it? This approach causes issues as our base FileStorage does not have a `data_folder`, so if we ever decide to use any other place for storing files this approach won't work.

Have you considered using the approach described [here](https://dash.plotly.com/dash-core-components/download#:~:text=DOWNLOAD%20TEXT-,Downloading%20a%20Dataframe%20as%20a%20CSV%20file,-For%20downloading%20dataframes)?

e.g.
```py
return dcc.send_data_frame(echo_bmg_combined_df.to_csv, filename)
```

This way you do not have to have this complex save/delete interaction with a separate thread

azywot replied:
Makes sense, will do and delete unnecessary filestorage functions

Tremirre commented on dashboard/storage/local.py:
```diff
@@ -29,3 +31,19 @@ def save_file(self, name: str, content: bytes) -> None:
             f.write(content)
+
+    def delete_file(self, name: str, seconds: int = 0) -> None:
+        time.sleep(seconds)
```
If you want to maintain the sleeping functionality I'd move it outside the `FileStorage` class so that if someone really wants to delete file with a sleep before hand he has to wrap it with a separate function (as putting it directly in the FileStorage class is quite unintuitive imo)

Tremirre reviewed (commented):
I added some suggestions to simplify the final download process,
but in In general code is good

As a sidenote I'd reconsider if we want to give that option of setting the exported filename or just fix it to some value with a timestamp (e.g. `screening_results_2023-07-16.csv`) as user may just as well rename the file after export

azywot commented:
> I added some suggestions to simplify the final download process, but in In general code is good
> 
> As a sidenote I'd reconsider if we want to give that option of setting the exported filename or just fix it to some value with a timestamp (e.g. `screening_results_2023-07-16.csv`) as user may just as well rename the file after export

Yeah makes sense, simplified it a bit so we've got sth like this right now:
![obraz](https://github.com/zuzg/drug-screening/assets/82370491/d3ce59d4-cad8-445e-80f1-68c523f071fe)

Tremirre reviewed (approved):
👌

zuzg reviewed (approved):
Nice! I will use it as input for correlation plots to see how it works in practice

AndrzejKaj reviewed (approved):
Looks good 😁
